server.js

```diff
@@ -2988,6 +2988,67 @@ cache(function(data, match, sendBadge, request) {
   })}
 ));
 
+// example https://img.shields.io/mingle/https/your-company.mingle-api.thoughtworks.com/your-project-name/card-number/your-property-name.svg
+// Mingle card status integration
+camp.route(/^\/mingle\/(http|https)\/(.*)\/(.*)\/(.*)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var scheme = match[1]; // http(s)
+  var host = match[2]; // your-company.mingle-api.thoughtworks.com
+  var project = match[3]; // your-project-name
+  var card = match[4]; // card-number e.g. `428`
+  var property = match[5]; // your-property-name e.g. `status`
+  var format = match[6];
+  var authToken = '';
+
+  var options = {
+    headers: {
+      'Content-Type': 'application/xml'
+    },
+    uri: scheme + '://' + host + '/api/v2/projects/' + project + '/cards/' + card + '.xml'
+  };
+
+  if (serverSecrets) {
+    // basic auth
+    options.headers.Authorization = 'Basic ' + new Buffer(serverSecrets.mingle_user + ':' + serverSecrets.mingle_password).toString('base64');
+  }
+
+  var badgeData = getBadgeData('mingle #' + card, data);
+
+  request(options, function(err, res, body) {
+    if (err != null || res.statusCode !== 200) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+
+    var xml2js = require('xml2js');
+
+    xml2js.parseString(body, function (err, result) {
+      if (err !== null) {
+        badgeData.text[1] = 'inaccessible';
+        sendBadge(format, badgeData);
+        return;
+      }
+
+      try {
+        var matchingProperty = result.card.properties[0].property.filter(function(p) {
+          return p.name[0].match(new RegExp(property,'i'));
```

Comment:
Please construct the regexp outside of the filter, to avoid re-compilations. Also, use `test` instead of `match` (ie, `regex.test(p.name[0])`.